easybuild/tools/include.py

```
@@ -0,0 +1,210 @@
+    init_path = os.path.join(path, '__init__.py')
+    try:
+        # note: can't use mkdir, since that required build options to be initialised
+        if not os.path.exists(path):
+            os.makedirs(path)
+
+        # put __init__.py files in place, with required pkgutil.extend_path statement
+        # note: can't use write_file, since that required build options to be initialised
+        handle = open(init_path, 'w')
+        handle.write(PKG_INIT_BODY)
+        handle.close()
+    except (IOError, OSError) as err:
+        raise EasyBuildError("Failed to write %s: %s", init_path, err)
+
+
+def set_up_eb_package(parent_path, eb_pkg_name, subpkgs=None):
+    """Set up new easybuild subnamespace in specified path."""
+    if not eb_pkg_name.startswith('easybuild'):
+        raise EasyBuildError("Specified EasyBuild package name does not start with 'easybuild': %s", eb_pkg_name)
+
+    pkgpath = os.path.join(parent_path, eb_pkg_name.replace('.', os.path.sep))
+
+    # handle subpackages first
+    if subpkgs:
+        for subpkg in subpkgs:
+            create_pkg(os.path.join(pkgpath, subpkg))
+
+    # creata package dirs on each level
+    while pkgpath != parent_path:
+        create_pkg(pkgpath)
+        pkgpath = os.path.dirname(pkgpath)
+
+
+def expand_glob_paths(glob_paths):
+    """Expand specified glob paths to a list of unique non-glob paths to only files."""
+    paths = []
+    for glob_path in glob_paths:
+        paths.extend([f for f in glob.glob(glob_path) if os.path.isfile(f)])
+
+    return nub(paths)
+
+
+def safe_symlink(source_path, symlink_path):
+    """Create a symlink at the specified path for the given path."""
+    try:
+        os.symlink(os.path.abspath(source_path), symlink_path)
+        _log.info("Symlinked %s to %s", source_path, symlink_path)
+    except OSError as err:
+        raise EasyBuildError("Symlinking %s to %s failed: %s", source_path, symlink_path, err)
```

But you are writing a library here, the 'user' you're talking about is another programmer writing a library or a binary.
A programmer won't be confused by a stacktrace, he will actually want one, so he can debug things.

When you write your binary you should check exceptions, and give meaningfull messages to users, but masking exceptions in a library just confuses other programmers.